2020/day9/solution.py

```diff
@@ -0,0 +1,33 @@
+
+def find_first_exception(numbers: list[int], lookback_count : int) -> int:
+    window = set(numbers[:lookback_count])
+
+    def window_sums_to_value(window, value):
+        return any((value - x in window and value - x != x) for x in window)
+
+    for i in range(lookback_count, len(numbers)):
+        if not window_sums_to_value(window, numbers[i]):
+            return numbers[i]
+        window.remove(numbers[i - lookback_count])
+        window.add(numbers[i])
```

Comment:
While you do handle not matching a number with itself above, what does the problem say about duplicate values in the list of numbers (really, duplicate numbers within a given window)?

This code looks fine for the case where there's never any dupes. Just for fun, how might you handle things if you did need to account for dupes?

Reply:
I think with dupes, you use a  Counter over the window, with key of number, and value of count.  Then verify that the count > 1 if it allows dupe in total